src/test/java/org/fcrepo/client/FcrepoResponseTest.java

```diff
@@ -69,4 +79,103 @@ public void testResponse() throws IOException {
                     "<http://example.org/book/3> <dc:title> \"Title\" .");
     }
 
+    /**
+     * Demonstrates that response objects are <em>not</em> {@code close()}ed by default, that the state of
+     * {@link FcrepoResponse#closed} is set appropriately when {@link FcrepoResponse#close()} is invoked under normal
+     * (i.e. no exception thrown during {@code close()}) conditions, and that {@link InputStream#close()} is not invoked
+     * repeatedly after the {@code FcrepoResponse} has been {@code close()}ed.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testClosableReleasesResources() throws Exception {
+        final InputStream mockBody = mock(InputStream.class);
+        final FcrepoResponse underTest = new FcrepoResponse(
+                URI.create("http://localhost/foo"), 201, "text/plain", URI.create("http://localhost/bar"), mockBody);
+
+        assertFalse("FcrepoResponse objects should not be closed until close() is invoked.", underTest.isClosed());
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+
+        underTest.close();
+        assertTrue(underTest.isClosed());
+        verify(mockBody, times(1)).close();
+    }
+
+    /**
+     * Demonstrates that if an {@code IOException} is thrown by {@link FcrepoResponse#close()}, <em>and</em> an
+     * exception is thrown inside of a client's {@code try} block, the {@code IOException} from the {@code close()}
+     * method is properly appended as a suppressed exception.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void testClosableSuppressedExceptions() throws Exception {
+        final InputStream mockBody = mock(InputStream.class);
+        final String notSuppressedMsg = "Not suppressed.";
+        final String suppressedMsg = "Suppressed IOE";
+        doThrow(new IOException(suppressedMsg)).when(mockBody).close();
+
+        try (FcrepoResponse underTest = new FcrepoResponse(URI.create("http://localhost/foo"), 201, "text/plain",
+                URI.create("http://localhost/bar"), mockBody)) {
+            assertFalse(underTest.isClosed());
+
+            throw new RuntimeException(notSuppressedMsg);
+
+        } catch (Exception e) {
+            assertEquals(notSuppressedMsg, e.getMessage());
```

Comment:
Instead of comparing messages between two constructed objects, you might throw a specific object from the mock and use a direct `==` comparison. I think that might be a bit clearer.